### PR TITLE
Fix build on Gentoo Linux

### PR DESCRIPTION
**Pull request by pasis**
Gentoo Linux has rolling updates and thus provides latest toolchain versions. This patchset contains of a subset of the patches required to build Motr on Gentoo system with latest software versions. It doesn't contain patches conflicting with #523 and should be treated as addition to that PR.

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

**Review by madhavemuri — APPROVED** [review]

File: motr/cob.c [hunk 1]
```diff
@@ -512,10 +512,13 @@ static void cob_fail_op(struct m0_op *op, int rc)
...
+		/* fallthrough */
 	case M0_ES_CREATING:
 		m0_sm_move(&op->op_entity->en_sm, 0, M0_ES_OPEN);
+		/* fallthrough */
 	case M0_ES_OPEN:
 		m0_sm_move(&op->op_entity->en_sm, 0, M0_ES_CLOSING);
+		/* fallthrough */
 	default:
 		m0_sm_move(&op->op_entity->en_sm, 0, M0_ES_INIT);
 	}
```
Comment (andriytk) [inline motr/cob.c:524]: @nikitadanilov should we really fallthrough here?

**Review by andriytk — APPROVED** [review]

**Review by madhavemuri — APPROVED** [review]

**Comment by huanghua78** [comment]
https://github.com/Seagate/cortx-motr/pull/523 is merged.

File: lib/user_space/timer.c [hunk 2]
```diff
@@ -67,8 +67,8 @@ M0_TL_DEFINE(tid, static, struct m0_timer_tid);
    gettid(2) implementation.
    Thread-safe, async-signal-safe.
  */
-static pid_t gettid() {
-
+static pid_t m0_gettid()
```
Comment (huanghua78) [inline lib/user_space/timer.c:70]: What function name is this gettid() conflicting with?
The system call gettid()?

Usually m0_ prefix is used for M0_INTERNAL or M0_EXTERNAL.
For static function, we don't use the m0_ prefix.

Can we change it to _gettid()?

**Comment by huanghua78** [comment]
@pasis can you please provide a OVA with Gentoo Linux for evaluation?

File: lib/user_space/timer.c (hunk 2 above)
Comment (pasis) [reply lib/user_space/timer.c]: Done.

Comment (pasis) [reply lib/user_space/timer.c]: And yes, it conflicts with glibc version of gettid().

**Comment by pasis** [comment]
> @pasis can you please provide a OVA with Gentoo Linux for evaluation?

@huanghua78, I use gentoo as my main system and I've built it from scratch (gentoo installation differs from binary distributions). The only image I know is here https://www.osboxes.org/gentoo/. But the image is old and requires updating the system what can be an issue for non-gentoo users. So I'm afraid I cannot help with the image.

**Review by huanghua78 — APPROVED** [review]

**Comment by huanghua78** [comment]
OK to test

**Pull request merged**